src/fs/filesystem.ts

```diff
@@ -624,86 +735,99 @@ export class FileSystem {
   // --------
 
   /** @internal */
-  async runOnNode<a>(
-    path: DistinctivePath,
-    isMutation: boolean,
-    fn: (node: Tree | File, relPath: Path) => Promise<a>
-  ): Promise<a> {
-    const parts = pathing.unwrap(path)
-    const head = parts[0]
-    const relPath = parts.slice(1)
-
-    const operation = isMutation
-      ? "make changes to"
-      : "query"
+  async checkMutationPermissionAndAddProof(path: DistinctivePath, isMutation: boolean): Promise<void> {
+    const operation = isMutation ? "make changes to" : "query"
 
     if (!this.localOnly) {
       const proof = await ucan.dictionary.lookupFilesystemUcan(path)
       const decodedProof = proof && ucan.decode(proof)
 
       if (!proof || !decodedProof || ucan.isExpired(decodedProof) || !decodedProof.signature) {
-        throw new NoPermissionError(`I don't have the necessary permissions to ${operation} the file system at "${pathing.toPosix(path)}"`)
+        throw new NoPermissionError(`I don't have the necessary permissions ${operation} to the file system at "${pathing.toPosix(path)}"`)
```

Comment:
Should stay as is I think? Change would result in "permissions make changes to to"

Reply:
Oh wow. Thanks for catching that. Must've been some multi-cursor thing messing this up!